### PR TITLE
Add Python type hints and doctests to other/two_sum.py

### LINKED ISSUE #2465: Add Python type hints and doctests to other/two_sum.py
@tonydelanuez Would you be willing to add Python type hints and doctests to [`other/two_sum.py`](../blob/master/other/two_sum.py)?  The function comments say that it will return a `list[int]` but instead it either returns a `list[list[int]]` or `False`.

### PR DESCRIPTION
**Pull request by spamegg1**
Fixes #2465

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/two_sum.py [hunk 1]
```diff
@@ -11,21 +11,36 @@
 Because nums[0] + nums[1] = 2 + 7 = 9,
 return [0, 1].
 """
+from typing import List
```
Comment (cclauss) [inline other/two_sum.py:14]:
```suggestion
from __future__ import annotations
```

File: other/two_sum.py [hunk 2]
```diff
@@ -11,21 +11,36 @@
...
 
 
-def twoSum(nums, target):
+def two_sum(nums: List[int], target: int) -> List[int]:
```
Comment (cclauss) [inline other/two_sum.py:17]:
```suggestion
def two_sum(nums: list[int], target: int) -> list[int]:
```

File: other/two_sum.py [hunk 3]
```diff
@@ -11,21 +11,36 @@
...
-    return False
+            return [chk_map[compl], index]
+        chk_map[val] = index
+    return []
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
```
Comment (cclauss) [inline other/two_sum.py:46]:
```suggestion
    doctest.testmod()
    print(f"{two_sum([2, 7, 11, 15], 9) = }")
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Changes to align with #2464 and show the user some results when they run the script.

File: other/two_sum.py (hunk 2 above)
Comment (cclauss) [reply other/two_sum.py:17]: Travis tests fail without this change.

Comment (spamegg1) [reply other/two_sum.py]: I clicked "Commit suggestion" but I guess it didn't work.

Comment (spamegg1) [reply other/two_sum.py]: I committed it manually below, still waiting for Travis to kick in.

Comment (cclauss) [reply other/two_sum.py]: You can `Add suggestion to batch` on all suggestions that then they will commit all at once.

**Review by cclauss — APPROVED** [review]

File: other/two_sum.py (hunk 2 above)
Comment (spamegg1) [reply other/two_sum.py]: Oh! Didn't know that! Thanks. Github is so confusing! It should be good now.

**Comment by spamegg1** [comment]
Thanks a lot @cclauss !

**Pull request merged**